src/createDeeplink.ts

```diff
@@ -0,0 +1,118 @@
+import { environment } from "@raycast/api";
+
+export type DeeplinkType = "extension" | "script-command"
+
+export type CreateScriptCommandDeeplinkOptions = {
+  /**
+   * The type of deeplink, which should be "script-command".
+   */
+  type: "script-command",
+  /**
+   * The name of the command.
+   */
+  command: string,
+  /**
+   * If the command accepts arguments, they can be passed using this query parameter.
+   */
+  arguments?: string[],
+};
+
+/**
+ * Options for creating a deeplink to an extension.
+ */
+export type CreateExtensionDeeplinkOptions = {
+  /**
+   * The type of deeplink, which should be "extension".
+   */
+  type: "extension",
+  /**
+   * The name of the owner or author of the extension.
+   */
+  ownerOrAuthorName: string,
+  /**
+   * The name of the extension.
+   */
+  extensionName: string,
```

Comment:
I would make it optional. If it is not defined you can get the extension name from the environment.

Reply:
I was thinking to keep everything required so that there would be no confusion in whether a command was referring to an external command, or one in the same extension.

Reply:
I got your point but why the lack of the `extensionName` would be confusing? If the is no `extensionName` prop it can only be from the same extension. I typically make deeplinks from the running extension command more often then an external extension.

Just some thoughts 😁.

Reply:
Without explicitly writing it out, I think people might accidentally miss it

Reply:
But, I think that not writing both `ownerOrAuthorName` and `extensionName` would be suitable. Unfortunately, `environment` currently doesn't include `ownerOrAuthorName` for some reason

Reply:
@erics118 You may need to update your `@raycast/api` to latest. See https://raycastcommunity.slack.com/archives/C02HEMAF2SJ/p1720007588026329.

Reply:
Seems like I missed that. Thanks!